arelle/plugin/validate/ESEF/__init__.py

```diff
@@ -696,11 +696,11 @@ def checkFootnote(elt: ModelInlineFootnote | ModelResource, text: str) -> None:
         if contextsWithPeriodTime:
             modelXbrl.error("ESEF.2.1.2.periodWithTimeContent",
                 _("The xbrli:startDate, xbrli:endDate and xbrli:instant elements MUST identify periods using whole days (i.e. specified without a time content): %(contextIds)s"),
-                modelObject=contextsWithPeriodTime, contextIds=", ".join(c.id for c in contextsWithPeriodTime))
+                modelObject=contextsWithPeriodTime, contextIds=", ".join(c.id for c in contextsWithPeriodTime if c.id is not None))
```

Comment:
`id` uses `get`, which by default returns a value and `None` so we need to introduce a guard against `None` here (or don't use get)

Reply:
You can assert these IDs are not None. [Contexts must have an ID](https://www.xbrl.org/Specification/XBRL-2.1/REC-2003-12-31/XBRL-2.1-REC-2003-12-31+corrected-errata-2013-02-20.html#_4.7.1) and they're coming from `ModelXbrl.contexts` which is a dict of contexts by ID. If they really don't have an ID they would fail to be included in the map these are built from.